### ISSUE TITLE
Restarting the round via vote while the round is ending causes improper restart

### ISSUE THREAD

**Issue by TimrodDX** [space-wizards/space-station-14#3984]
I think I've finally figured out what makes this happen and I haven't seen any reports about it, so..

How to reproduce:
- Call the shuttle and wait until the shuttle has nearly arrived (10-15 seconds).
- Call a restart vote and vote yes so the restart vote goes off at the same time as the shuttle's arrival

The server will then go into this weird state where the round will end, it'll dump everyone back to the lobby, but won't properly reset the map. Instead, it'll respawn people onto the map from last round and the map will not be reset until an admin restarts the server.

**Comment by 20kdc** [comment]
This was observed on Lizard [US West] at around 2021-05-12 19:09:22 BST.
Something else to note is that the round start timer is instant in this event.
![image](https://user-images.githubusercontent.com/22304167/118048082-ab363700-b373-11eb-8c24-b183e99379dc.png)

**Comment by AJCM-git** [comment]
Probably fixed after moony's `restartround` changes, hopefully (?

**Comment by moonheart08** [comment]
@AJCM-git No, not fixed, seems if it's timed right you can still achieve the double-restart and brick things exactly as described.
